locale.c

```diff
@@ -2251,10 +2251,11 @@ S_querylocale_2008_i(pTHX_ const locale_category_index index,
              * system distinugishes "C" from "POSIX", we do too.) */
             if (cur_obj == PL_C_locale_obj && ! isNAME_C_OR_POSIX(retval)) {
                 Safefree(retval);
-                retval = savepv("C");
+                retval = "C";
+            }
+            else {
+                SAVEFREEPV(retval);
             }
```

Comment:
The test `cur_obj == PL_C_locale_obj && ! isNAME_C_OR_POSIX(retval)` seems very cheap to me, certainly a lot cheaper than savepv(), wouldn't it be better to do this test within the lock and avoid the copy entirely?